### ISSUE TITLE
Syntax highlight doesn't always kick in properly

### ISSUE THREAD

**Issue by darkhog** [engor/Jentos_IDE#42]
Example: https://i.imgur.com/q9GCZbO.png

Currently seen this only in dark themes, however I'm not sure if the problem is related. Needs investigation. It's a minor bug, one I can live with, however I'd like it fixed if possible.

//edit: Nope, unrelated to the theme, switched back to QT Creator and the same thing kept happening. However I've noticed that when I've started editing stuff on the lines with broken highlighting or around it, it reappeared. Does Jentos not recolor the whole file on each edit? If not, sounds like an easy fix I mean, it would be slow for large files, but you should kept your code modular anyway.

Of course it'd be better to find out why it breaks in the first place, but if it can't be done, this fix could work.

**Comment by engor** [comment]
Theme switching is unfinished a bit. To fix editor area - go to Preferences - Editor - Reset theme colors (link).
IDE don't reset editor colors because you can changed them in prefs.
I know that ide should store user's colors separately for each theme, but... When I was going to dev app the days of monkey-x gone to end.
Don't know will I fix that or not.
So, use solution written above for fixing colors.

**Comment by darkhog** [comment]
No, I meant syntax highlight itself, see the picture, not colors and textures of buttons and so on. Also all dark themes (and light ones for that matter) seems to be identical. Why not just have two themes, one light and one dark?

**Comment by darkhog** [comment]
The thing is, part of the code seems like it was colored like a comment or not colored at all and where at some point syntax highlighting suddenly starts to work as it should. See the picture.

**Comment by engor** [comment]
![2017-11-15_17-12-48](https://user-images.githubusercontent.com/5233803/32830641-6f0101b2-ca28-11e7-8c0d-25e08e4120fa.png)

**Comment by darkhog** [comment]
I know what you meant when you wrote it first time. However I don't think you realize what did I mean. I've meant that the part of the code is unhighlighted while at some unspecified line syntax highlight starts working as expected, not that the theme is using editor colors from an other theme. I've defined my own editor colors based on Turbo Pascal/QBasic syntax highlight anyway.

**Comment by engor** [comment]
Oops.. Hm, you get unhighlighted area just after theme was changes? Reopen doc fix that?

PS. you can drag-n-drop dochable areas to re-arrange them - Projects / Source / Code Tree / etc.

**Comment by darkhog** [comment]
Well, already rearranged the docks, but thanks. Anyway, I really dunno how did I trigger the highlighter bug. It seems to be doing the thing on a line-per-line basis and (for the specific line) seems to go away after I edit a bit like add a space or whatever (for that line, rest is still unhighlighted).

The easiest thing would be to force-highlight the entire document. It would be slower the larger file gets, but you shouldn't put all of your code in a single file anyway for readability and maintenance purposes.

I mean, the best case would be to see what is actually causing the problem and breaks the highlighter, but force highlighting of the entire file would work as well.

Really puzzling bug, since I didn't see anything like that in Ted itself, but yeah. The best thing I can tell you is that comments may have something to do with it since in cases where I got highlighter broken there was usually a comment nearby and the unhighlighted code seemed to use the same color that is used for comments.

In the big picture, it doesn't matter much since it is purely visual bug that I can live with, but yeah.

Anyway, any chances of you going back to developing Jentos further now that MX is back in development thanks to Cerberus team? Other IDEs seems to be either prohibitively expensive (Jungle) or underpowered compared to Jentos, still better than Ted though.

//edit: Also I don't think theme changing triggers the bug now that I think about it. Messing with themes sometimes affects it, but not always

**Comment by darkhog** [comment]
Okay, seem to have it. After closing and reopening the file (not always though) the highlight bug can be triggered. What's interesting is that the mere fact of opening the Options dialog (not even doing anything in it, just opening it) seem to fix the issue. Weird.

**Comment by darkhog** [comment]
Made a bug showcase. https://youtu.be/wLpaOoHj7UU Video's choppy because my computer is crap and because I've used CamStudio which is crap on its own.
